<!DOCTYPE html>
<html lang="zh-CN">
<head>
  <meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=2">
<meta name="theme-color" content="#222">
<meta name="generator" content="Hexo 5.4.0">
  <link rel="apple-touch-icon" sizes="180x180" href="/images/apple-touch-icon-next.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/images/favicon-32x32-next.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/images/favicon-16x16-next.png">
  <link rel="mask-icon" href="/images/logo.svg" color="#222">

<link rel="stylesheet" href="/css/main.css">

<link rel="stylesheet" href="https://fonts.loli.net/css?family=Noto Serif SC:300,300italic,400,400italic,700,700italic&display=swap&subset=latin,latin-ext">
<link rel="stylesheet" href="/lib/font-awesome/css/all.min.css">
  <link rel="stylesheet" href="//cdn.jsdelivr.net/npm/pace-js@1/themes/blue/pace-theme-minimal.css">
  <script src="//cdn.jsdelivr.net/npm/pace-js@1/pace.min.js"></script>

<script id="hexo-configurations">
    var NexT = window.NexT || {};
    var CONFIG = {"hostname":"benettlee.github.io","root":"/","scheme":"Gemini","version":"7.8.0","exturl":false,"sidebar":{"position":"left","display":"post","padding":18,"offset":12,"onmobile":false},"copycode":{"enable":true,"show_result":true,"style":"mac"},"back2top":{"enable":true,"sidebar":false,"scrollpercent":false},"bookmark":{"enable":false,"color":"#222","save":"auto"},"fancybox":false,"mediumzoom":false,"lazyload":false,"pangu":false,"comments":{"style":"tabs","active":null,"storage":true,"lazyload":false,"nav":null},"algolia":{"hits":{"per_page":10},"labels":{"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}},"localsearch":{"enable":true,"trigger":"auto","top_n_per_article":1,"unescape":false,"preload":false},"motion":{"enable":true,"async":false,"transition":{"post_block":"fadeIn","post_header":"slideDownIn","post_body":"slideDownIn","coll_header":"slideLeftIn","sidebar":"slideUpIn"}},"path":"search.xml"};
  </script>

  <meta name="description" content="linux å¸¸ç”¨å‘½ä»¤123456789# 1ã€å°†windowsä¸‹ç¼–è¾‘æ–‡ä»¶è½¬æ¢ä¸ºlinuxæ–‡ä»¶ï¼Œé˜²æ­¢å‘ç”Ÿé”™è¯¯dos2unix merged.txtdos2unix: æ­£åœ¨è½¬æ¢æ–‡ä»¶ merged.txt ä¸ºUnixæ ¼å¼...# 2ã€æŸ¥çœ‹linuxå½“å‰ç”¨æˆ·w# 3ã€æŸ¥çœ‹æ–‡ä»¶å¤¹ä¸‹çš„æ–‡ä»¶æ•°ç›®ls -l |grep &quot;^-&quot;|wc -læˆ–find .&#x2F;folder_name -type f">
<meta property="og:type" content="article">
<meta property="og:title" content="RNAseqåˆ†ææµç¨‹ï¼ˆBYMEï¼‰">
<meta property="og:url" content="https://benettlee.github.io/2021/05/30/N9AxXFYknJwvXM7D/index.html">
<meta property="og:site_name" content="BenettLee Blog">
<meta property="og:description" content="linux å¸¸ç”¨å‘½ä»¤123456789# 1ã€å°†windowsä¸‹ç¼–è¾‘æ–‡ä»¶è½¬æ¢ä¸ºlinuxæ–‡ä»¶ï¼Œé˜²æ­¢å‘ç”Ÿé”™è¯¯dos2unix merged.txtdos2unix: æ­£åœ¨è½¬æ¢æ–‡ä»¶ merged.txt ä¸ºUnixæ ¼å¼...# 2ã€æŸ¥çœ‹linuxå½“å‰ç”¨æˆ·w# 3ã€æŸ¥çœ‹æ–‡ä»¶å¤¹ä¸‹çš„æ–‡ä»¶æ•°ç›®ls -l |grep &quot;^-&quot;|wc -læˆ–find .&#x2F;folder_name -type f">
<meta property="og:locale" content="zh_CN">
<meta property="og:image" content="https://gitee.com/benettlee/images/raw/master/img/f31d26f1.jpg">
<meta property="og:image" content="https://gitee.com/benettlee/images/raw/master/img/385879e1.png">
<meta property="og:image" content="https://gitee.com/benettlee/images/raw/master/img/8a444a31.png">
<meta property="article:published_time" content="2021-05-30T11:54:47.000Z">
<meta property="article:modified_time" content="2021-05-30T11:59:10.507Z">
<meta property="article:author" content="æé¹ç¨‹">
<meta name="twitter:card" content="summary">
<meta name="twitter:image" content="https://gitee.com/benettlee/images/raw/master/img/f31d26f1.jpg">

<link rel="canonical" href="https://benettlee.github.io/2021/05/30/N9AxXFYknJwvXM7D/">


<script id="page-configurations">
  // https://hexo.io/docs/variables.html
  CONFIG.page = {
    sidebar: "",
    isHome : false,
    isPost : true,
    lang   : 'zh-CN'
  };
</script>

  <title>RNAseqåˆ†ææµç¨‹ï¼ˆBYMEï¼‰ | BenettLee Blog</title>
  






  <noscript>
  <style>
  .use-motion .brand,
  .use-motion .menu-item,
  .sidebar-inner,
  .use-motion .post-block,
  .use-motion .pagination,
  .use-motion .comments,
  .use-motion .post-header,
  .use-motion .post-body,
  .use-motion .collection-header { opacity: initial; }

  .use-motion .site-title,
  .use-motion .site-subtitle {
    opacity: initial;
    top: initial;
  }

  .use-motion .logo-line-before i { left: initial; }
  .use-motion .logo-line-after i { right: initial; }
  </style>
</noscript>

<link rel="alternate" href="/atom.xml" title="BenettLee Blog" type="application/atom+xml">
</head>

<body itemscope itemtype="http://schema.org/WebPage">
  <div class="container use-motion">
    <div class="headband"></div>

    <header class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-container">
  <div class="site-nav-toggle">
    <div class="toggle" aria-label="åˆ‡æ¢å¯¼èˆªæ ">
      <span class="toggle-line toggle-line-first"></span>
      <span class="toggle-line toggle-line-middle"></span>
      <span class="toggle-line toggle-line-last"></span>
    </div>
  </div>

  <div class="site-meta">

    <a href="/" class="brand" rel="start">
      <span class="logo-line-before"><i></i></span>
      <h1 class="site-title">BenettLee Blog</h1>
      <span class="logo-line-after"><i></i></span>
    </a>
      <p class="site-subtitle" itemprop="description">å¥½è®°æ€§ä¸å¦‚çƒ‚ç¬”å¤´</p>
  </div>

  <div class="site-nav-right">
    <div class="toggle popup-trigger">
        <i class="fa fa-search fa-fw fa-lg"></i>
    </div>
  </div>
</div>




<nav class="site-nav">
  <ul id="menu" class="main-menu menu">
        <li class="menu-item menu-item-home">

    <a href="/" rel="section"><i class="fa fa-home fa-fw"></i>é¦–é¡µ</a>

  </li>
        <li class="menu-item menu-item-about">

    <a href="/about/" rel="section"><i class="fa fa-user fa-fw"></i>å…³äº</a>

  </li>
        <li class="menu-item menu-item-tags">

    <a href="/tags/" rel="section"><i class="fa fa-tags fa-fw"></i>æ ‡ç­¾<span class="badge">12</span></a>

  </li>
        <li class="menu-item menu-item-categories">

    <a href="/categories/" rel="section"><i class="fa fa-th fa-fw"></i>åˆ†ç±»<span class="badge">6</span></a>

  </li>
        <li class="menu-item menu-item-archives">

    <a href="/archives/" rel="section"><i class="fa fa-archive fa-fw"></i>å½’æ¡£<span class="badge">28</span></a>

  </li>
        <li class="menu-item menu-item-commonweal">

    <a href="/404/" rel="section"><i class="fa fa-heartbeat fa-fw"></i>å…¬ç›Š 404</a>

  </li>
      <li class="menu-item menu-item-search">
        <a role="button" class="popup-trigger"><i class="fa fa-search fa-fw"></i>æœç´¢
        </a>
      </li>
  </ul>
</nav>



  <div class="search-pop-overlay">
    <div class="popup search-popup">
        <div class="search-header">
  <span class="search-icon">
    <i class="fa fa-search"></i>
  </span>
  <div class="search-input-container">
    <input autocomplete="off" autocapitalize="off"
           placeholder="æœç´¢..." spellcheck="false"
           type="search" class="search-input">
  </div>
  <span class="popup-btn-close">
    <i class="fa fa-times-circle"></i>
  </span>
</div>
<div id="search-result">
  <div id="no-result">
    <i class="fa fa-spinner fa-pulse fa-5x fa-fw"></i>
  </div>
</div>

    </div>
  </div>

</div>
    </header>

    
  <div class="back-to-top">
    <i class="fa fa-arrow-up"></i>
    <span>0%</span>
  </div>
  <div class="reading-progress-bar"></div>


    <main class="main">
      <div class="main-inner">
        <div class="content-wrap">
          

          <div class="content post posts-expand">
            

    
  
  
  <article itemscope itemtype="http://schema.org/Article" class="post-block" lang="zh-CN">
    <link itemprop="mainEntityOfPage" href="https://benettlee.github.io/2021/05/30/N9AxXFYknJwvXM7D/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="image" content="/images/tx.jpg">
      <meta itemprop="name" content="æé¹ç¨‹">
      <meta itemprop="description" content="ä¸€å®šè¦åšæœ€é€‚åˆè‡ªå·±çš„äº‹æƒ…ï¼Œä¸è¦è¿åˆåˆ«äººçš„å£å‘³è€Œå»åšä¸€ä»¶ä¸å±äºè‡ªæˆ‘çš„â€œéš¾äº‹â€ã€‚">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="BenettLee Blog">
    </span>
      <header class="post-header">
        <h1 class="post-title" itemprop="name headline">
          RNAseqåˆ†ææµç¨‹ï¼ˆBYMEï¼‰
        </h1>

        <div class="post-meta">
            <span class="post-meta-item">
              <span class="post-meta-item-icon">
                <i class="far fa-calendar"></i>
              </span>
              <span class="post-meta-item-text">å‘è¡¨äº</span>
              

              <time title="åˆ›å»ºæ—¶é—´ï¼š2021-05-30 11:54:47 / ä¿®æ”¹æ—¶é—´ï¼š11:59:10" itemprop="dateCreated datePublished" datetime="2021-05-30T11:54:47Z">2021-05-30</time>
            </span>

          <br>
            <span class="post-meta-item" title="æœ¬æ–‡å­—æ•°">
              <span class="post-meta-item-icon">
                <i class="far fa-file-word"></i>
              </span>
                <span class="post-meta-item-text">æœ¬æ–‡å­—æ•°ï¼š</span>
              <span>15k</span>
            </span>
            <span class="post-meta-item" title="é˜…è¯»æ—¶é•¿">
              <span class="post-meta-item-icon">
                <i class="far fa-clock"></i>
              </span>
                <span class="post-meta-item-text">é˜…è¯»æ—¶é•¿ &asymp;</span>
              <span>14 åˆ†é’Ÿ</span>
            </span>

        </div>
      </header>

    
    
    
    <div class="post-body" itemprop="articleBody">

      
        <h3 id="linux-å¸¸ç”¨å‘½ä»¤"><a href="#linux-å¸¸ç”¨å‘½ä»¤" class="headerlink" title="linux å¸¸ç”¨å‘½ä»¤"></a>linux å¸¸ç”¨å‘½ä»¤</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># 1ã€å°†windowsä¸‹ç¼–è¾‘æ–‡ä»¶è½¬æ¢ä¸ºlinuxæ–‡ä»¶ï¼Œé˜²æ­¢å‘ç”Ÿé”™è¯¯</span></span><br><span class="line">dos2unix merged.txt</span><br><span class="line">dos2unix: æ­£åœ¨è½¬æ¢æ–‡ä»¶ merged.txt ä¸ºUnixæ ¼å¼...</span><br><span class="line"><span class="comment"># 2ã€æŸ¥çœ‹linuxå½“å‰ç”¨æˆ·</span></span><br><span class="line">w</span><br><span class="line"><span class="comment"># 3ã€æŸ¥çœ‹æ–‡ä»¶å¤¹ä¸‹çš„æ–‡ä»¶æ•°ç›®</span></span><br><span class="line">ls -l |grep <span class="string">&quot;^-&quot;</span>|wc -l</span><br><span class="line">æˆ–</span><br><span class="line">find ./folder_name -<span class="built_in">type</span> f | wc -l     æ³¨ï¼šfolder_name æŒ‡çš„æ˜¯ï¼šå½“å‰ç›®å½•ä¸‹çš„ä½ è¦æŸ¥è¯¢çš„æ–‡ä»¶å¤¹å</span><br></pre></td></tr></table></figure>

<h2 id="RNAseqç®€å•åˆ†ææµç¨‹"><a href="#RNAseqç®€å•åˆ†ææµç¨‹" class="headerlink" title="RNAseqç®€å•åˆ†ææµç¨‹"></a>RNAseqç®€å•åˆ†ææµç¨‹</h2><h3 id="install-softwares"><a href="#install-softwares" class="headerlink" title="install softwares"></a>install softwares</h3><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line">conda install -c bioconda -n RNAseq samtools </span><br><span class="line">conda install -c bioconda -n RNAseq multiqc</span><br><span class="line">conda install -c bioconda -n RNAseq hisat2</span><br><span class="line">conda install -c bioconda -n RNAseq cufflinks</span><br><span class="line">conda install -c bioconda -n RNAseq taco</span><br><span class="line">conda install -c bioconda -n RNAseq HTSeq </span><br><span class="line">conda install -c bioconda -n RNAseq subread <span class="comment">##ä½¿ç”¨featureCounts</span></span><br><span class="line">conda install -n RNAseq fastp</span><br><span class="line">conda install -c bioconda -n RNAseq gffcompare</span><br><span class="line">conda install -c bioconda -n RNAseq rmats rmats2sashimiplot</span><br></pre></td></tr></table></figure>

<h3 id="raw-dataçš„è¿‡æ»¤å’Œæ¸…é™¤ä¸å¯ä¿¡æ•°æ®ï¼ˆclean-readsï¼‰"><a href="#raw-dataçš„è¿‡æ»¤å’Œæ¸…é™¤ä¸å¯ä¿¡æ•°æ®ï¼ˆclean-readsï¼‰" class="headerlink" title="raw dataçš„è¿‡æ»¤å’Œæ¸…é™¤ä¸å¯ä¿¡æ•°æ®ï¼ˆclean readsï¼‰"></a>raw dataçš„è¿‡æ»¤å’Œæ¸…é™¤ä¸å¯ä¿¡æ•°æ®ï¼ˆclean readsï¼‰</h3><h3 id="ç”¨fastpå¯¹è½¬å½•ç»„æ•°æ®åšQC"><a href="#ç”¨fastpå¯¹è½¬å½•ç»„æ•°æ®åšQC" class="headerlink" title="ç”¨fastpå¯¹è½¬å½•ç»„æ•°æ®åšQC"></a>ç”¨fastpå¯¹è½¬å½•ç»„æ•°æ®åšQC</h3><ul>
<li>å¯¹æ•°æ®è‡ªåŠ¨è¿›è¡Œå…¨æ–¹ä½è´¨æ§ï¼Œç”Ÿæˆäººæ€§åŒ–çš„æŠ¥å‘Šï¼›</li>
<li>è¿‡æ»¤åŠŸèƒ½(ä½è´¨é‡,å¤ªçŸ­,å¤ªå¤šNâ€¦â€¦)ï¼›</li>
<li>å¯¹æ¯ä¸€ä¸ªåºåˆ—çš„å¤´éƒ¨æˆ–å°¾éƒ¨ï¼Œè®¡ç®—æ»‘åŠ¨çª—å†…çš„è´¨é‡å‡å€¼ï¼Œå¹¶å°†å‡å€¼è¾ƒä½çš„å­åºåˆ—è¿›è¡Œåˆ‡é™¤(ç±»ä¼¼ Trimmomatic çš„åšæ³•,ä½†æ˜¯å¿«éå¸¸å¤š)ï¼›</li>
<li>å…¨å±€å‰ªè£ (åœ¨å¤´/å°¾éƒ¨ï¼Œä¸å½±å“å»é‡)ï¼Œå¯¹äº Illumina ä¸‹æœºæ•°æ®å¾€å¾€æœ€åä¸€åˆ°ä¸¤ä¸ª cycle éœ€è¦è¿™æ ·å¤„ç†ï¼›</li>
<li>å»é™¤æ¥å¤´æ±¡æŸ“ã€‚å‰å®³çš„æ˜¯ï¼Œä½ ä¸ç”¨è¾“å…¥æ¥å¤´åºåˆ—ï¼Œå› ä¸ºç®—æ³•ä¼šè‡ªåŠ¨è¯†åˆ«æ¥å¤´åºåˆ—å¹¶è¿›è¡Œå‰ªè£ï¼›</li>
<li>å¯¹äºåŒç«¯æµ‹åº(PE)çš„æ•°æ®ï¼Œè½¯ä»¶ä¼šè‡ªåŠ¨æŸ¥æ‰¾æ¯ä¸€å¯¹readçš„é‡å åŒºåŸŸï¼Œå¹¶å¯¹è¯¥é‡å åŒºåŸŸä¸­ä¸åŒ¹é…çš„ç¢±åŸºå¯¹è¿›è¡Œæ ¡æ­£ï¼›</li>
<li>å»é™¤å°¾éƒ¨çš„ polyGã€‚å¯¹äºNextSeq/NovaSeq çš„æµ‹åºæ•°æ®ï¼Œå› ä¸ºæ˜¯ä¸¤è‰²æ³•å‘å…‰ï¼ŒpolyG æ˜¯å¸¸æœ‰çš„äº‹ï¼Œæ‰€ä»¥è¯¥ç‰¹æ€§å¯¹è¯¥ä¸¤ç±»æµ‹åºå¹³å°é»˜è®¤æ‰“å¼€ï¼›</li>
<li>å¯¹äºPEæ•°æ®ä¸­çš„overlapåŒºé—´ä¸­ä¸ä¸€è‡´çš„ç¢±åŸºå¯¹ï¼Œä¾æ®è´¨é‡å€¼è¿›è¡Œæ ¡æ­£ï¼›</li>
<li>å¯ä»¥å¯¹å¸¦åˆ†å­æ ‡ç­¾(UMI)çš„æ•°æ®è¿›è¡Œé¢„å¤„ç†ï¼Œä¸ç®¡UMIåœ¨æ’å…¥ç‰‡æ®µè¿˜æ˜¯åœ¨index ä¸Šï¼Œéƒ½å¯ä»¥è½»æ¾å¤„ç†ï¼›</li>
<li>å¯ä»¥å°†è¾“å‡ºè¿›è¡Œåˆ†æ‹†ï¼Œè€Œä¸”æ”¯æŒä¸¤ç§æ¨¡å¼ï¼Œåˆ†åˆ«æ˜¯æŒ‡å®šåˆ†æ‹†çš„ä¸ªæ•°ï¼Œæˆ–è€…åˆ†æ‹†åæ¯ä¸ªæ–‡ä»¶çš„è¡Œæ•°ã€‚</li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##å•ç«¯æµ‹åº</span></span><br><span class="line">fastp -i in.fq -o out.fq</span><br><span class="line"><span class="comment">## åŒç«¯æµ‹åº</span></span><br><span class="line">fastp -w 5 -h *.html -j *.json -i in.R1.fq -o out.R1.fq -I in.R2.fq -O out.R2.fq</span><br></pre></td></tr></table></figure>

<h3 id="Step1ï¼šå»ºç«‹ç´¢å¼•ï¼ŒæŠŠreads-mappingåˆ°åŸºå› ç»„ä¸Š"><a href="#Step1ï¼šå»ºç«‹ç´¢å¼•ï¼ŒæŠŠreads-mappingåˆ°åŸºå› ç»„ä¸Š" class="headerlink" title="Step1ï¼šå»ºç«‹ç´¢å¼•ï¼ŒæŠŠreads mappingåˆ°åŸºå› ç»„ä¸Š"></a>Step1ï¼šå»ºç«‹ç´¢å¼•ï¼ŒæŠŠreads mappingåˆ°åŸºå› ç»„ä¸Š</h3><ul>
<li><p><strong>å»ºç´¢å¼•</strong></p>
<blockquote>
<p><strong>1ã€å°†gff3è½¬æ¢ä¸ºgtf</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">gffread -T *.gff3 -o *.gtf</span><br><span class="line"><span class="comment">##for example</span></span><br><span class="line">gffread -T /mnt/sdb/data/bioinfo/genome/SolTub_3.0/Solanum_tuberosum.SolTub_3.0.45.gff3 -o SolTub_3.0.gtf</span><br></pre></td></tr></table></figure>

<p><strong>2ã€æå– .exon æ–‡ä»¶</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">extract_exons.py *.gtf &gt;*.exon</span><br><span class="line"><span class="comment">##for example</span></span><br><span class="line">python ~/soft/anaconda3/envs/RNAseq/bin/hisat2_extract_exons.py SolTub_3.0.gtf &gt; SolTub_3.0.exon</span><br></pre></td></tr></table></figure>

<p><strong>3ã€æå– .ss æ–‡ä»¶</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">extract_splice_sites.py *.gtf &gt;*.ss</span><br><span class="line"><span class="comment">##for example</span></span><br><span class="line">python ~/soft/anaconda3/envs/RNAseq/bin/hisat2_extract_splice_sites.py SolTub_3.0.gtf &gt; SolTub_3.0.ss</span><br></pre></td></tr></table></figure>

<p><strong>4ã€ç”¨ä»¥ä¸Šæ–‡ä»¶å»ºç«‹ç´¢å¼•</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">hisat2-build -p 20 --ss *.ss --exon *.exon *.fa indexname</span><br><span class="line"><span class="comment">###faæ–‡ä»¶ä¸ºåŸºå› ç»„æ–‡ä»¶</span></span><br><span class="line"><span class="comment">##for example</span></span><br><span class="line">hisat2-build -p 30 --ss SolTub_3.0.ss --exon SolTub_3.0.exon /mnt/sdb/data/bioinfo/genome/SolTub_3.0/SolTub_3.0.dna.toplevel.fa ../index/SolTub</span><br></pre></td></tr></table></figure></blockquote>
</li>
<li><p><strong>å°†reads mappingåˆ°åŸºå› ç»„ä¸Š</strong></p>
<blockquote>
<p><strong>1ã€hisat2</strong><br><strong>hisatéé“¾ç‰¹å¼‚æ€§</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">hisat2 -x [index] --dta-cufflinks  -p 4  -1 *.r1.fq.gz -2 *.r2.fq.gz -S output.sam</span><br></pre></td></tr></table></figure>

<p><strong>é“¾ç‰¹å¼‚æ€§</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">hisat2 -x [index] --dta-cufflinks --rna-strandness RF -p 4  -1 *.r1.fq.gz -2 *.r2.fq.gz -S output.sam  | tee -a *.<span class="built_in">log</span></span><br><span class="line"><span class="keyword">for</span> example</span><br><span class="line">hisat2 -x ../index/SolTub --dta-cufflinks --rna-strandness RF -p 20 -1 ../raw/AtC_0h_rep1_R1.fq.gz -2 ../raw/AtC_0h_rep1_R2.fq.gz -S AtC_0h_rep1.sam | tee log.AtC_0h_rep1.sam</span><br></pre></td></tr></table></figure>

<p><strong>2ã€æŸ¥çœ‹æ¯”å¯¹ç‡</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##æ–¹æ³•1ï¼šæŸ¥çœ‹log</span></span><br><span class="line"><span class="comment">##æ–¹æ³•2ï¼šç”¨ samtools çš„ flagstat è¿›è¡Œç»Ÿè®¡ï¼Œä¹Ÿèƒ½å¾—åˆ°æ¯”å¯¹ç‡</span></span><br><span class="line">samtools flagstat RNA.bam</span><br></pre></td></tr></table></figure>

<p><strong>3ã€è¿‡æ»¤æ‰mapping reads å‚æ•°Q30ï¼Œ æœ‰äº›åŸºå› åºåˆ—ç›¸ä¼¼åº¦å¤ªé«˜ï¼Œmapping readså°±ä¸å¤ªå‡†ï¼Œæ‰€ä»¥è¿‡æ»¤æ‰è¿™äº›åºåˆ—ç›¸ä¼¼åº¦è¿‡é«˜çš„readsã€‚</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">samtools view</span><br><span class="line">samtools view -@ 20 -Sb -q 30 <span class="variable">$sam</span> -o <span class="variable">$bam</span></span><br></pre></td></tr></table></figure>

<p><strong>4ã€å¦‚æœæ˜¯<code>paired-end</code>æ•°æ®å¿…é¡»æŒ‰ç…§readsåç§°æ’åº</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">samtools sort -@ 20 <span class="variable">$bam</span> -o <span class="variable">$sorted</span>.bam</span><br></pre></td></tr></table></figure></blockquote>
</li>
</ul>
<h3 id="Step2ï¼šè½¬å½•æœ¬ç»„è£…ä¸æ‹¼æ¥ï¼ˆcufflinksæˆ–è€…StringTieï¼‰"><a href="#Step2ï¼šè½¬å½•æœ¬ç»„è£…ä¸æ‹¼æ¥ï¼ˆcufflinksæˆ–è€…StringTieï¼‰" class="headerlink" title="Step2ï¼šè½¬å½•æœ¬ç»„è£…ä¸æ‹¼æ¥ï¼ˆcufflinksæˆ–è€…StringTieï¼‰"></a>Step2ï¼šè½¬å½•æœ¬ç»„è£…ä¸æ‹¼æ¥ï¼ˆcufflinksæˆ–è€…StringTieï¼‰</h3><ul>
<li><strong>cufflinks</strong></li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#é“¾ç‰¹å¼‚æ€§</span></span><br><span class="line">cufflinks  -p 4 --library-type fr-firststrand -g reference.gff -o outdir  sorted.bam </span><br><span class="line"><span class="comment">##for example</span></span><br><span class="line">cufflinks  -p 30 --library-type fr-firststrand -g /mnt/sdb/data/bioinfo/genome/SolTub_3.0/SolTub_3.0.45.gff3 -o ../cufflinks/QsC_0h_rep1 QsC_0h_rep1.sorted.bam</span><br></pre></td></tr></table></figure>

<ul>
<li><strong>StringTie</strong><br><strong>StringTie ä½¿ç”¨è¯´æ˜:</strong></li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br></pre></td><td class="code"><pre><span class="line">stringtie &lt;input.bam ..&gt; [-G &lt;guide_gff&gt;] [-l &lt;label&gt;] [-o &lt;out_gtf&gt;] [-p &lt;cpus&gt;]</span><br><span class="line">  [-v] [-a &lt;min_anchor_len&gt;] [-m &lt;min_tlen&gt;] [-j &lt;min_anchor_cov&gt;] [-f &lt;min_iso&gt;]</span><br><span class="line">  [-C &lt;coverage_file_name&gt;] [-c &lt;min_bundle_cov&gt;] [-g &lt;bdist&gt;] [-u]</span><br><span class="line">  [-e] [-x &lt;seqid,..&gt;] [-A &lt;gene_abund.out&gt;] [-h] &#123;-B | -b &lt;dir_path&gt;&#125;</span><br><span class="line"><span class="comment">#for example</span></span><br><span class="line"><span class="comment">### é“¾ç‰¹å¼‚æ€§æ–‡åº“</span></span><br><span class="line">stringtie --rf -p 8 -G chrX.gtf sample.sort.bam -o sample.gtf</span><br></pre></td></tr></table></figure>

<p><strong>é€‰é¡¹:</strong></p>
<blockquote>
<p><strong>â€“version : è¾“å‡ºè½¯ä»¶çš„ç‰ˆæœ¬ä¿¡æ¯<br>-G å‚è€ƒåºåˆ—çš„åŸºå› æ³¨é‡Šæ–‡ä»¶ (GTF/GFF3)<br>-l è¾“å‡ºè½¬å½•æœ¬çš„åç§°å‰ç¼€ (default: STRG)<br>-f æœ€å°‘è½¬å½•æœ¬çš„æ¯”ä¾‹ (default: 0.1)<br>-m ç»„è£…è½¬å½•æœ¬çš„æœ€å°é•¿åº¦ (default: 200)<br>-o ç»„è£…è½¬å½•æœ¬çš„GTFæ³¨é‡Šæ–‡ä»¶ (default: stdout)<br>-a è¿æ¥ä½ç‚¹é”šå®šåºåˆ—çš„æœ€å°é•¿åº¦ (default: 10)<br>-j è¿æ¥ä½ç‚¹çš„æœ€å°è¦†ç›–åº¦ (default: 1)<br>-t åŸºäºè¦†ç›–åº¦å¯¹é¢„æµ‹çš„è½¬å½•æœ¬è¿›è¡Œä¿®æ­£ (default: coverage trimming is enabled)<br>-c ç»„è£…è½¬å½•æœ¬çš„readsæœ€å°è¦†ç›–åº¦(default: 2.5)<br>-v è¾“å‡ºlog ä¿¡æ¯<br>-g æ¯”å¯¹ä¸Šçš„reads é—´è·å¤§äºé˜€å€¼åˆ™æ–°åŸä¸€ä¸ªæ–°çš„è½¬å½•æŸ (default: 50)<br>-C è¾“å‡ºå‚è€ƒè½¬å½•æœ¬ä¸­è¢«reads è¦†ç›–åˆ°çš„è½¬å½•æœ¬<br>-M è½¬å½•æŸå…è®¸å¤šæ¯”å¯¹readsè¦†ç›–çš„æœ€å¤§å æ¯” (default:0.95)<br>-p çº¿ç¨‹ï¼ˆCPU)æ•° (default: 1)<br>-A åŸºå› ä¸°éƒ½è¾“å‡ºæ–‡ä»¶<br>-B åœ¨è¾“å‡ºçš„GFTåŒç›®å½•ä¸‹è¾“å‡ºBallgown table æ–‡ä»¶<br>-b åœ¨ &lt;dir_path&gt; ç›®å½•ä¸‹è¾“å‡ºBallgown table æ–‡ä»¶<br>-e åªå¯¹å‚è€ƒè½¬å½•æœ¬è¿›è¡Œä¸°éƒ½è¯„ä¼° (requires -G)<br>-x ä¸åœ¨å‚è€ƒåºåˆ—åŒºåŸŸç»„è£…ä»»ä½•çš„æ–°è½¬å½•æœ¬<br>-u å¤šæ¯”å¯¹æ ¡æ­£ (default: correction enabled)<br>-h è¾“å‡ºè½¯ä»¶çš„å¸®åŠ©ä¿¡æ¯</strong></p>
</blockquote>
<p><strong>è½¬å½•æœ¬åˆå¹¶æ¨¡å¼ä½¿ç”¨è¯´æ˜:</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">stringtie --merge [Options] &#123; gtf_list | strg1.gtf ...&#125;</span><br><span class="line"><span class="comment">## for example</span></span><br><span class="line">stringtie --merge -p 8 -G chrX.gtf -o stringtie_merge.gtf mergelist.txt</span><br></pre></td></tr></table></figure>

<p><strong>é€‰é¡¹</strong></p>
<blockquote>
<p><strong>-G &lt;guide_gff&gt;   å‚è€ƒè½¬å½•æœ¬çš„æ³¨é‡Šä¿¡æ¯ (GTF/GFF3)<br>-o &lt;out_gtf&gt;     åˆå¹¶è½¬å½•æœ¬çš„GTFè¾“å‡ºæ–‡ä»¶ (default: stdout)<br>-m &lt;min_len&gt;     åˆå¹¶è½¬å½•æœ¬çš„æœ€å°é•¿åº¦(default: 50)<br>-c &lt;min_cov&gt;     åˆå¹¶è½¬å½•æœ¬çš„æœ€ä½è¦†ç›–åº¦(default: 0)<br>-F &lt;min_fpkm&gt;    åˆå¹¶è½¬å½•æœ¬çš„æœ€å°FPKMå€¼(default: 1.0)<br>-T &lt;min_tpm&gt;     åˆå¹¶è½¬å½•æœ¬çš„æœ€å°TPMå€¼(default: 1.0)<br>-f &lt;min_iso&gt;     isoform æœ€å°æ¯”ä¾‹(default: 0.01)<br>-g &lt;gap_len&gt;     è½¬å½•æœ¬è§GAPé•¿åº¦å°äºé˜€å€¼åˆ™åˆå¹¶ä¸¤è½¬å½•æœ¬ (default: 250)<br>-i               å…è®¸åˆå¹¶è½¬å½•æœ¬ä¸­æœ‰å†…å«å­ä¿ç•™; by default<br>-l <label>       è¾“å‡ºçš„è½¬å½•æœ¬åç§°å‰ç¼€ (default: MSTRG)</strong></p>
</blockquote>
<h3 id="Step3ï¼šè®¡ç®—countsæ•°"><a href="#Step3ï¼šè®¡ç®—countsæ•°" class="headerlink" title="Step3ï¼šè®¡ç®—countsæ•°"></a>Step3ï¼šè®¡ç®—countsæ•°</h3><p><strong>1ã€ç»è¿‡cufflinksä¹‹åæ¯ä¸ªæ ·éƒ½ä¼šäº§ç”Ÿä¸€ä¸ªæ–‡ä»¶å¤¹ï¼Œæ¯ä¸ªæ–‡ä»¶å¤¹é‡Œéƒ½æœ‰ä¸€ä¸ªtranscript.gtf, éœ€è¦åœ¨è¿è¡Œä¸‹ä¸€æ­¥ä¹‹å‰å°†æ‰€æœ‰çš„transcript.gtfæ”¾åˆ°ä¸€ä¸ªæ–‡ä»¶é‡Œgtflist</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">ls dir/*/transcripts.gtf &gt;gtflist</span><br><span class="line"><span class="comment">## for example</span></span><br><span class="line">ls cufflinks/*/transcripts.gtf &gt;gtflist</span><br></pre></td></tr></table></figure>

<p><strong>2ã€cuffmergeä»£æ›¿taco</strong></p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">cuffmerge -p 20  -s <span class="variable">$genome</span> -g <span class="variable">$gtf</span> gtflist &gt;log.cuffmerge.txt</span><br><span class="line"><span class="comment">##for example</span></span><br><span class="line">cuffmerge -p 20  -s /mnt/sdb/data/bioinfo/genome/SolTub_3.0/SolTub_3.0.dna.toplevel.fa -g ../process/SolTub_3.0.gtf gtflist &gt;AQ0cuffmerge.log</span><br></pre></td></tr></table></figure>

<p><strong>3 è®¡ç®—countæ•°</strong></p>
<ul>
<li><strong>3.1 htseq-countè®¡ç®—countæ•°</strong></li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##é“¾ç‰¹å¼‚æ€§</span></span><br><span class="line">htseq-count -f bam -q  -s reverse sorted.bam  assembly.gtf  &gt;output.txt 2&gt;&amp;1  &amp;</span><br><span class="line"><span class="comment">### for example </span></span><br><span class="line">htseq-count -f bam -q  -s reverse ../sam/AtC_0h_rep1.sorted.bam  merged.gtf  &gt;AtC_0h_rep1_count.txt 2&gt;&amp;1  &amp;</span><br></pre></td></tr></table></figure>

<ul>
<li><strong>3.2 æ–°æ‹¼è£…è½¬å½•æœ¬ä¸åŸåŸºå› ç»„æ³¨é‡Šè½¬å½•æœ¬æ¯”è¾ƒ (å¯ç”¨æ¥ç­›é€‰æ–°è½¬å½•æœ¬)</strong></li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># -R: åªè€ƒè™‘ä¸æ‹¼è£…çš„è½¬å½•æœ¬æœ‰é‡å çš„æ³¨é‡Šè½¬å½•æœ¬</span></span><br><span class="line"><span class="comment"># -r: reference gtf</span></span><br><span class="line"><span class="comment"># -o: è¾“å‡ºå‰ç¼€</span></span><br><span class="line"></span><br><span class="line"><span class="built_in">cd</span> ~/transcriptome/data</span><br><span class="line">gffcompare -R -r genome/GRCh38.gtf -o assembeCompare2Ref ehbio_trans.gtf</span><br><span class="line"><span class="comment"># ä¼šè¾“å‡ºä¸€ä¸ªassembeCompare2Ref.annotated.gtfï¼Œç”¨äºåç»­çš„å®šé‡</span></span><br><span class="line"></span><br><span class="line">ç»Ÿè®¡ä¸åŒç§ç±»è½¬å½•æœ¬çš„æ•°ç›®</span><br><span class="line">cut -f 3 assembeCompare2Ref.ehbio_trans.gtf.tmap | tail -n +2 | sort | uniq -c</span><br></pre></td></tr></table></figure>

<ul>
<li><strong>3.3 featureCountsè®¡ç®—countæ•°</strong></li>
</ul>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">#é“¾ç‰¹å¼‚æ€§</span></span><br><span class="line">featureCounts -T 5 -s 2 -t exon -g gene_id -a annotation.gtf -o counts.txt .bam</span><br><span class="line"><span class="comment">##for example</span></span><br><span class="line">featureCounts -T 20 -s 2 -p -t exon -g gene_id -a merged.gtf -o AtC_0h_rep_count.txt ../sam/AtC_0h_rep1.sorted.bam</span><br></pre></td></tr></table></figure>

<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment">##æ¯ä¸ªåŸºå› ä¸Šçš„countæ•°ï¼Œåˆ™åªéœ€è¦æå–å‡ºç¬¬1åˆ—å’Œç¬¬7åˆ—çš„ä¿¡æ¯</span></span><br><span class="line">cut -f 1,7 *featureCounts.txt |grep -v <span class="string">&#x27;^#&#x27;</span> &gt;feacturCounts.txt</span><br></pre></td></tr></table></figure>

<h3 id="å·®å¼‚è¡¨è¾¾åˆ†æ"><a href="#å·®å¼‚è¡¨è¾¾åˆ†æ" class="headerlink" title="å·®å¼‚è¡¨è¾¾åˆ†æ"></a>å·®å¼‚è¡¨è¾¾åˆ†æ</h3><p><a href="!%5B%5D(https://gitee.com/benettlee/images/raw/master/img/DESeq2.zip)">DEseq2åˆ†æè„šæœ¬</a></p>
<h3 id="rMATSå·®å¼‚å¯å˜å‰ªåˆ‡åˆ†æ"><a href="#rMATSå·®å¼‚å¯å˜å‰ªåˆ‡åˆ†æ" class="headerlink" title="rMATSå·®å¼‚å¯å˜å‰ªåˆ‡åˆ†æ"></a><strong>rMATSå·®å¼‚å¯å˜å‰ªåˆ‡åˆ†æ</strong></h3><p>rMATSæ˜¯ä¸€æ¬¾å¯¹RNA-Seqæ•°æ®è¿›è¡Œå·®å¼‚å¯å˜å‰ªåˆ‡åˆ†æçš„è½¯ä»¶ã€‚å…¶é€šè¿‡rMATSç»Ÿè®¡æ¨¡å‹å¯¹ä¸åŒæ ·æœ¬ï¼ˆæœ‰ç”Ÿç‰©å­¦é‡å¤çš„ï¼‰è¿›è¡Œå¯å˜å‰ªåˆ‡äº‹ä»¶çš„è¡¨è¾¾å®šé‡ï¼Œç„¶åä»¥likelihood-ratio testè®¡ç®—P valueæ¥è¡¨ç¤ºä¸¤ç»„æ ·å“åœ¨IncLevelï¼ˆInclusion Levelï¼‰æ°´å¹³ä¸Šçš„å·®å¼‚(ä»å…¬å¼ä¸Šæ¥çœ‹ï¼ŒIncLevelè·ŸPSIçš„å®šä¹‰ä¹Ÿæ˜¯ç±»ä¼¼çš„)ï¼ŒlncLevelå¹¶åˆ©ç”¨Benjamini Hochbergç®—æ³•å¯¹p valueè¿›è¡Œæ ¡æ­£å¾—FDRå€¼ã€‚rMATSå¯è¯†åˆ«çš„å¯å˜å‰ªåˆ‡äº‹ä»¶æœ‰5ç§ï¼Œåˆ†åˆ«æ˜¯skipped exon (SE)å¤–æ˜¾å­è·³è·ƒï¼Œalternative 5â€™ splice site (A5SS)ç¬¬ä¸€ä¸ªå¤–æ˜¾å­å¯å˜å‰ªåˆ‡ï¼Œalternative 3â€™ splice site (A3SS)æœ€åä¸€ä¸ªå¤–æ˜¾å­å¯å˜å‰ªåˆ‡ï¼Œmutually exclusive exons (MXE)å¤–æ˜¾å­é€‰æ‹©æ€§è·³è·ƒå’Œ retained intron (RI)å†…å«å­æ»ç•™ï¼Œå±•ç°å½¢å¼å¦‚ä¸‹å›¾ï¼ˆæ¥è‡ªå®˜ç½‘<a target="_blank" rel="noopener external nofollow noreferrer" href="http://rnaseq-mats.sourceforge.net/index.html">Multivariate Analysis of Transcript Splicing (MATS)</a>ï¼‰<br><img src="https://gitee.com/benettlee/images/raw/master/img/f31d26f1.jpg" alt="rMATS_splicing.jpg"></p>
<ul>
<li><strong>è½¯ä»¶ä¸‹è½½åŠå®‰è£…</strong><br>rMATSæœ€è¿‘åˆšç°åœ¨å‡ºäº†rMATS 4.0.1ç‰ˆï¼Œç›¸æ¯”ä¹‹é—´çš„rMATS 3.2.5ç‰ˆï¼Œå…¶ç”¨Cï¼ŒPythonï¼ŒCythoné‡å†™äº†è¯¥è½¯ä»¶ï¼Œè¿ç®—é€Ÿåº¦æå‡äº†100å€ï¼Œå¹¶ä¸”å¯æ”¯æŒå¤šçº¿ç¨‹æ‰§è¡Œï¼ˆæ˜æ˜¾æ„Ÿè§‰åˆ°è®¡ç®—é€Ÿåº¦çš„æå‡ï¼‰ï¼Œå¹¶ä¸”æ–°ç‰ˆçš„å®‰è£…ä¹Ÿç®€ä¾¿å¥½å¤šäº†ã€‚PS.è€ç‰ˆçš„rMATSæˆ‘é‚£æ—¶éƒ½æ˜¯ç”¨biocondaå®‰è£…çš„ï¼Œä¸ç„¶å¤ªæŠ˜è…¾äº†ã€‚ã€‚</li>
</ul>
<p>è¿›å®˜ç½‘æˆ–è€…ä¸‹è¿°ç½‘ç«™è¿›è¡Œä¸‹è½½<br><a target="_blank" rel="noopener external nofollow noreferrer" href="https://sourceforge.net/projects/rnaseq-mats/files/MATS/rMATS.4.0.1.tgz/download">Download RNASeq-MATS from SourceForge.net</a><br>ç„¶åæŒ‰ç…§å®˜ç½‘è¯´æ˜å®‰è£…ä¸€äº›åº“æ–‡ä»¶ä»¥åŠpythonåº“ï¼ˆä»¥ubuntuä¸ºä¾‹ï¼‰</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br></pre></td><td class="code"><pre><span class="line">pip install numpy</span><br><span class="line">sudo apt-get install libblas-dev liblapack-dev</span><br><span class="line">sudo apt-get install gfortran</span><br></pre></td></tr></table></figure>

<p>å¦‚æœpythonçš„numpyè£…çš„æœ‰é—®é¢˜ï¼Œå¯ä»¥ä½¿ç”¨biocondaæ¥è£…ä¸‹æ—§ç‰ˆçš„rMATSï¼Œå…¶ä¼šé¡ºä¾¿æŠŠnumpyä¹Ÿè£…å¥½ï¼Œç„¶åå°†å…¶æ”¾ç½®åœ¨ç¯å¢ƒå˜é‡ä¸­å°±è¡Œäº†ï¼ˆä¸€èˆ¬ä¹Ÿä¸ç”¨è¿™æ ·ï¼‰</p>
<p>å¦‚æœè¿è¡Œæ—¶æŠ¥é”™ï¼šerror while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directoryï¼Œåˆ™è¯´æ˜ç¼ºå°‘libgsl.so.0åº“æ–‡ä»¶ï¼ŒæŒ‰ç…§ä¸‹è¿°å®‰è£…ä¸‹å°±å¥½</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">sudo apt-get install libgsl0-dev</span><br></pre></td></tr></table></figure>

<ul>
<li><strong>è½¯ä»¶çš„ä½¿ç”¨</strong><br>è½¯ä»¶ä½¿ç”¨ä¹Ÿå¾ˆç®€å•ï¼ŒrMATSæ”¯æŒä¸¤ç§æ ¼å¼æ–‡ä»¶çš„è¾“å…¥ã€‚ç¬¬ä¸€ç§æ˜¯fastqæ ¼å¼ï¼Œé‚£ä¹ˆåœ¨å®‰è£…çš„æ—¶å€™è¿˜éœ€è¦å®‰è£…STARæ¯”å¯¹è½¯ä»¶ä»¥åŠæä¾›æ¯”å¯¹çš„ç´¢å¼•æ–‡ä»¶ï¼ˆSTARçš„ç´¢å¼•æ–‡ä»¶å¼‚å¸¸çš„å¤§ï¼‰ï¼Œæ‰€ä»¥rMATSå…¶å®æ˜¯å»ºè®®ä½¿ç”¨ç¬¬äºŒç§æ–¹å¼ï¼›ç¬¬äºŒç§æ˜¯bamæ ¼å¼ï¼ŒrMATSæ”¯æŒå…¶ä»–æ¯”å¯¹è½¯ä»¶æ¯”å¯¹åçš„ç»“æœbamæ–‡ä»¶ä½œä¸ºè¾“å…¥ï¼Œæ¯”å¦‚tophatç­‰ï¼ˆé‚£ä¹ˆhisat2ä¹Ÿæ²¡å•¥é—®é¢˜ï¼Œæˆ‘è¯•è¿‡ï¼‰ï¼Œè¿™æ ·ä¹Ÿèƒ½å‡å°‘rMATSçš„è¿è¡Œæ—¶é—´ã€‚</li>
</ul>
<p>æ–°ç‰ˆrMATSä¸‹è½½è§£å‹åï¼Œä½ ä¼šå‘ç°æœ‰ä¸¤ä¸ªrmats.pyæ‰§è¡Œè„šæœ¬ï¼Œè¿™æ˜¯ç”±äºrMATS v4.0.1 (turbo) was built with two different settings of Python interpreterï¼Œæ‰€ä»¥æˆ‘ä»¬éœ€è¦å…ˆæµ‹è¯•ä¸‹è‡ªå·±çš„ç³»ç»Ÿæ”¯æŒé‚£ç§ï¼Œè¿›å…¥pythonï¼Œè¾“å…¥ä¸‹è¿°å‘½ä»¤</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br></pre></td><td class="code"><pre><span class="line">&gt;&gt;&gt; import sys</span><br><span class="line">&gt;&gt;&gt; <span class="built_in">print</span> sys.maxunicode</span><br></pre></td></tr></table></figure>

<p>å¦‚æœå‡ºç°1114111åˆ™è¯´æ˜éœ€è¦ä½¿ç”¨<code>rMATS-turbo-Linux-UCS4</code>æ–‡ä»¶å¤¹ä¸‹<code>rmats.py</code>ï¼›å¦‚æœå‡ºç°<code>65535</code>åˆ™è¯´æ˜ä½¿ç”¨<code>rMATS-turbo-Linux-UCS2</code>æ–‡ä»¶å¤¹ä¸‹<code>rmats.py</code></p>
<p>rMATSçš„å‚æ•°è®¾ç½®ä¸å¤šï¼Œæˆ‘ä¸€èˆ¬ä½¿ç”¨ä»¥ä¸‹å‡ ä¸ªï¼Œå…¶ä»–å…·ä½“å¯å‚è€ƒå®˜ç½‘</p>
<blockquote>
<p>â€“b1 b1.txt è¾“å…¥sample1çš„txtæ ¼å¼çš„æ–‡ä»¶ï¼Œæ–‡ä»¶å†…ä»¥é€—å·åˆ†éš”é‡å¤æ ·æœ¬çš„bamæ–‡ä»¶å<br>â€“b2 b2.txt è¾“å…¥sample2çš„txtæ ¼å¼çš„æ–‡ä»¶ï¼Œæ–‡ä»¶å†…ä»¥é€—å·åˆ†éš”é‡å¤æ ·æœ¬çš„bamæ–‡ä»¶å<br>-t readType åŒç«¯æµ‹åºåˆ™readTypeä¸ºpairedï¼Œå•ç«¯æµ‹åºåˆ™ä¸ºsingle<br>â€“readLength æµ‹åºreadsçš„é•¿åº¦<br>â€“gtf gtfFile éœ€è¦è¾“å…¥çš„gtfæ–‡ä»¶<br>â€“od outDir æ‰€æœ‰è¾“å‡ºæ–‡ä»¶çš„è·¯å¾„ï¼ˆæ–‡ä»¶å¤¹ï¼‰<br>â€“nthread è®¾ç½®çº¿ç¨‹æ•°<br>â€“cstat The cutoff splicing difference. The cutoff used in the null hypothesis test for differential splicingï¼ˆè¿™ä¸ªæˆ‘ä¸€ç›´ä¸å¤ªç†è§£æ˜¯æ€ä¹ˆå¡çš„é˜ˆå€¼ï¼Œæ˜¯åœ¨ç®—æ³•è¯†åˆ«ä¸€äº›æ–°çš„å¯å˜å‰ªåˆ‡çš„æ—¶å€™çš„å·®å¼‚æ€§å—ï¼‰</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">python rMATS-turbo-Linux-UCS4/rmats.py --b1 b1.txt --b2 b2.txt --gtf Homo_sapiens.Ensembl.GRCh37.72.gtf --od AS -t paired --readLength 151 --cstat 0.0001 --nthread 10</span><br></pre></td></tr></table></figure>
</blockquote>
<ul>
<li><strong>ç»“æœæ–‡ä»¶</strong><br>rMATSçš„ç»“æœæ–‡ä»¶æ˜¯ä»¥å„ä¸ªå¯å˜å‰ªåˆ‡äº‹ä»¶çš„åˆ†å¸ƒçš„ï¼Œä¸»è¦ç”±AS_Event.MATS.JC.txtï¼ŒAS_Event.MATS.JCEC.txtï¼ŒfromGTF.AS_Event.txtï¼ŒJC.raw.input.AS_Event.txtï¼ŒJCEC.raw.input.AS_Event.txtè¿™å‡ ç±»ï¼›å…¶ä¸­JCå’ŒJCECçš„åŒºåˆ«åœ¨äºå‰è€…è€ƒè™‘è·¨è¶Šå‰ªåˆ‡ä½ç‚¹çš„readsï¼Œè€Œåè€…ä¸ä»…è€ƒè™‘å‰è€…çš„readsè¿˜è€ƒè™‘åˆ°åªæ¯”å¯¹åˆ°ç¬¬ä¸€å¼ å›¾ä¸­æ¡çº¹çš„åŒºåŸŸï¼ˆä¹Ÿå°±æ˜¯è¯´æ²¡æœ‰è·¨è¶Šå‰ªåˆ‡ä½ç‚¹çš„readsï¼‰ï¼Œä½†æ˜¯æˆ‘ä»¬ä¸€èˆ¬ä½¿ç”¨JCçš„ç»“æœå°±å¤Ÿäº†ï¼ˆå¦‚æœåªæ˜¯å•çº¯çš„æ¯”è¾ƒä¸¤ç»„æ ·å“é—´å¯å˜å‰ªåˆ‡çš„å·®å¼‚çš„è¯ï¼‰ã€‚</li>
</ul>
<p>è¿™å‡ ç±»æ–‡ä»¶ä¸­æ¯”è¾ƒé‡è¦çš„è¦æ•°S_Event.MATS.JC.txtï¼Œå› ä¸ºå…¶ä»–æ–‡ä»¶çš„ä¿¡æ¯å·®ä¸å¤šæœ€ç»ˆéƒ½æ•´åˆåœ¨è¿™ä¸ªæ–‡ä»¶é‡Œé¢ï¼Œä»¥SE.MATS.JC.txtä¸ºä¾‹ï¼š</p>
<p>1-5åˆ—çœ‹åˆ—åå°±èƒ½æ‡‚å…¶æ„æ€çš„ï¼Œåˆ†åˆ«IDï¼ŒGeneIDï¼ŒgeneSymbolï¼Œchrï¼Œstrand</p>
<p>6-11åˆ—åˆ†åˆ«ä¸ºå¤–æ˜¾å­çš„ä½ç½®ä¿¡æ¯ï¼Œåˆ†åˆ«ä¸ºexonStart_0baseï¼ŒexonEndï¼ŒupstreamESï¼ŒupstreamEEï¼ŒdownstreamESï¼ŒdownstreamEEï¼›ç½‘ä¸Šæœ‰å¼ å›¾èƒ½å¾ˆå¥½çš„è§£é‡Šå…¶å«ä¹‰ï¼Œå¦‚ä¸‹æ‰€ç¤ºï¼›å…¶ä»–å¯å˜å‰ªåˆ‡æ–‡ä»¶çš„è¿™å‡ åˆ—æœ‰ç‚¹ç•¥å¾®ä¸åŒï¼Œä½†éƒ½å¯ä»¥ç±»ä¼¼çš„ç†è§£</p>
<p><img src="https://gitee.com/benettlee/images/raw/master/img/385879e1.png" alt="skipped_exon.png"></p>
<p>12åˆ—åˆä¸ºIDï¼Œä¸çŸ¥é“ä¸ºå•¥é‡å¤å†æ¥ä¸€æ¬¡ï¼Œå¯èƒ½ä¸ºäº†å¸ƒå±€ç¾è§‚å§ã€‚ã€‚ã€‚</p>
<p>13-16åˆ—å±•ç¤ºä¸¤ç»„æ ·å“åœ¨inclusion junctionï¼ˆIJCï¼‰å’Œskipping junction countsï¼ˆSJCï¼‰ä¸‹çš„countæ•°ï¼Œé‡å¤æ ·æœ¬çš„ç»“æœä»¥é€—å·åˆ†éš”ï¼›åˆ—ååˆ†åˆ«ä¸ºIJC_SAMPLE_1ï¼ŒSJC_SAMPLE_1ï¼ŒIJC_SAMPLE_2ï¼ŒSJC_SAMPLE_2ï¼Œå¯ä»¥ä»ä¸‹å›¾æ¥ç†è§£ä¸‹</p>
<p><img src="https://gitee.com/benettlee/images/raw/master/img/8a444a31.png" alt="SE.png"></p>
<p>ä¸‹é¢å‡ åˆ—ä¿¡æ¯rMATSè®¤ä¸ºæ˜¯æä¸ºé‡è¦çš„ä¿¡æ¯ï¼š</p>
<ul>
<li>lncFormLen ï¼šlength of inclusion form, used for normalization</li>
<li>SkipFormLen : length of skipping form, used for normalization</li>
<li>P-Value : Significance of splicing difference between two sample groupsï¼ˆä¸¤ç»„æ ·å“å¯å˜å‰ªåˆ‡çš„ç»Ÿè®¡å­¦æ˜¾è‘—å·®å¼‚æŒ‡æ ‡ï¼‰</li>
<li>FDR ï¼š False Discovery Rate calculated from p-valueï¼ˆå¯¹p-valueçš„FDRæ ¡æ­£ï¼‰</li>
<li>lncLevel1 ï¼š inclusion level for SAMPLE_1 replicates (comma separated) calculated from normalized counts</li>
<li>IncLevel2 : inclusion level for SAMPLE_2 replicates (comma separated) calculated from normalized counts</li>
<li>IncLevelDifference ï¼š average(IncLevel1) - average(IncLevel2)<br>lncFormLenå’ŒSkipFormLenåˆ†åˆ«æ˜¯inclusion formå’Œskipping formçš„æœ‰æ•ˆé•¿åº¦å€¼ï¼Œè™½ç„¶æœ‰è®¡ç®—å…¬å¼ä½†æ˜¯è¿˜æ˜¯è¦æ ¹æ®readsè·¨è¶Šæ—¶çš„å…·ä½“æƒ…å†µæ¥å®šçš„ï¼Œå…·ä½“è§£é‡Šå¯è§<a target="_blank" rel="noopener external nofollow noreferrer" href="https://groups.google.com/forum/#!topic/rmats-user-group/d7rzUBKXF1U">Google ç½‘ä¸Šè®ºå›</a>(éœ€ç¿»å¢™ã€‚ã€‚ã€‚æ‰èƒ½çœ‹å“¦)</li>
</ul>
<p>lncLevel1å’ŒIncLevel2åˆ†åˆ«ä¸ºsample1å’Œsample2çš„inclusion levelï¼ˆç²—ç•¥å¯ç†è§£ä¸ºPSIï¼‰ï¼Œä»å…¬å¼ä¸Šçœ‹æ¯”è¾ƒå¥½ç†è§£</p>
<figure class="highlight shell"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">Ïˆ = (I/LI) / (I/LI + S/LS)</span><br></pre></td></tr></table></figure>

<p><code>Ïˆ = Inclusion Levelï¼ˆsample1ï¼‰ï¼ŒI = IJC_SAMPLE_1ï¼ŒS = SJC_SAMPLE_1ï¼ŒLI = lncFormLenï¼ŒLS = SkipFormLen</code></p>
<p>ä»å…¬å¼ä¸Šç†è§£ä¸ºlncLevelæ˜¯åœ¨æœ€ç»ˆçš„æˆç†ŸmRNAä¸­ï¼Œè¿™ä¸ªrespective exonï¼ˆSEäº‹ä»¶ä¸­åˆ™æ˜¯å¯¹åº”é‚£ä¸ªè¢«è·³è·ƒçš„exonï¼‰å‡ºç°çš„å¹³å‡é¢‘ç‡æ˜¯å¤šå°‘ï¼Œæˆ–è€…è¯´æ‰€å çš„æ¯”ä¾‹ï¼Ÿä¹Ÿå¯ä»¥è®¤ä¸ºæ˜¯read counts(æ ‡å‡†åŒ–åçš„)åœ¨å¯å˜å‰ªåˆ‡çš„äº‹ä»¶ä¸­çš„å„è‡ªexonä¸Šæ‰€å çš„æ¯”ä¾‹ï¼Ÿç®€å•çš„è¯´å°±æ˜¯lncLevelè¶Šå°è¯´æ˜å‡ºç°å¤–æ˜¾å­çš„è·³è·ƒçš„æ¯”ä¾‹è¶Šé«˜ï¼ˆè¿™ä¸ªæ˜¯ä»¥SEæ–‡ä»¶ä¸ºä¾‹çš„ï¼‰ï¼Ÿï¼Ÿï¼Ÿå¯èƒ½æˆ‘è¿˜ç†è§£çš„ä¸æ˜¯å¾ˆå‡†ç¡®ã€‚ã€‚ã€‚ã€‚</p>
<p>æœ€åè¿˜æœ‰ä¸€ä¸ªè½¯ä»¶èƒ½ç»˜åˆ¶å¯å˜å‰ªåˆ‡çš„å›¾ç‰‡çš„è½¯ä»¶rmats2sashimiplot</p>
<h3 id="Getting-error-in-running-rMATS"><a href="#Getting-error-in-running-rMATS" class="headerlink" title="Getting error in running rMATS"></a>Getting error in running rMATS</h3><p>Hi everyone,</p>
<p>Iâ€™m trying to run rMATS to get alternative splicing but encountered some errors.</p>
<p>Errors found:</p>
<pre><code>==========
Done processing each gene from dictionary to compile AS events
Found 14962 exon skipping events
Found 1100 exon MX events
Found 28836 alt SS events
There are 16303 alt 3 SS events and 12533 alt 5 SS events.
Found 25486 RI events
==========

Running the statistical part.
/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe: error while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File &quot;/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_P/FDR.py&quot;, line 45, in &lt;module&gt;
    ifile=open(sys.argv[1]);title=ifile.readline();
IOError: [Errno 2] No such file or directory: &#39;/tmp/tmpQ7Q7J5/JC_SE/rMATS_result_P-V.txt&#39;
paste: /tmp/tmpQ7Q7J5/JC_SE/rMATS_result_FDR.txt: No such file or directory
/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe: error while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File &quot;/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_P/FDR.py&quot;, line 45, in &lt;module&gt;
    ifile=open(sys.argv[1]);title=ifile.readline();
IOError: [Errno 2] No such file or directory: &#39;/tmp/tmpQ7Q7J5/JCEC_SE/rMATS_result_P-V.txt&#39;
paste: /tmp/tmpQ7Q7J5/JCEC_SE/rMATS_result_FDR.txt: No such file or directory
/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe: error while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File &quot;/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_P/FDR.py&quot;, line 45, in &lt;module&gt;
    ifile=open(sys.argv[1]);title=ifile.readline();
IOError: [Errno 2] No such file or directory: &#39;/tmp/tmpQ7Q7J5/JC_MXE/rMATS_result_P-V.txt&#39;
paste: /tmp/tmpQ7Q7J5/JC_MXE/rMATS_result_FDR.txt: No such file or directory
/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe: error while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File &quot;/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_P/FDR.py&quot;, line 45, in &lt;module&gt;
    ifile=open(sys.argv[1]);title=ifile.readline();
IOError: [Errno 2] No such file or directory: &#39;/tmp/tmpQ7Q7J5/JCEC_MXE/rMATS_result_P-V.txt&#39;
paste: /tmp/tmpQ7Q7J5/JCEC_MXE/rMATS_result_FDR.txt: No such file or directory
/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe: error while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File &quot;/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_P/FDR.py&quot;, line 45, in &lt;module&gt;
    ifile=open(sys.argv[1]);title=ifile.readline();
IOError: [Errno 2] No such file or directory: &#39;/tmp/tmpQ7Q7J5/JC_A3SS/rMATS_result_P-V.txt&#39;
paste: /tmp/tmpQ7Q7J5/JC_A3SS/rMATS_result_FDR.txt: No such file or directory
/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe: error while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File &quot;/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_P/FDR.py&quot;, line 45, in &lt;module&gt;
    ifile=open(sys.argv[1]);title=ifile.readline();
IOError: [Errno 2] No such file or directory: &#39;/tmp/tmpQ7Q7J5/JCEC_A3SS/rMATS_result_P-V.txt&#39;
paste: /tmp/tmpQ7Q7J5/JCEC_A3SS/rMATS_result_FDR.txt: No such file or directory
/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe: error while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File &quot;/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_P/FDR.py&quot;, line 45, in &lt;module&gt;
    ifile=open(sys.argv[1]);title=ifile.readline();
IOError: [Errno 2] No such file or directory: &#39;/tmp/tmpQ7Q7J5/JC_A5SS/rMATS_result_P-V.txt&#39;
paste: /tmp/tmpQ7Q7J5/JC_A5SS/rMATS_result_FDR.txt: No such file or directory
/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe: error while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File &quot;/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_P/FDR.py&quot;, line 45, in &lt;module&gt;
    ifile=open(sys.argv[1]);title=ifile.readline();
IOError: [Errno 2] No such file or directory: &#39;/tmp/tmpQ7Q7J5/JCEC_A5SS/rMATS_result_P-V.txt&#39;
paste: /tmp/tmpQ7Q7J5/JCEC_A5SS/rMATS_result_FDR.txt: No such file or directory
/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe: error while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File &quot;/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_P/FDR.py&quot;, line 45, in &lt;module&gt;
    ifile=open(sys.argv[1]);title=ifile.readline();
IOError: [Errno 2] No such file or directory: &#39;/tmp/tmpQ7Q7J5/JC_RI/rMATS_result_P-V.txt&#39;
paste: /tmp/tmpQ7Q7J5/JC_RI/rMATS_result_FDR.txt: No such file or directory
/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe: error while loading shared libraries: libgsl.so.0: cannot open shared object file: No such file or directory
Traceback (most recent call last):
  File &quot;/home/ABZ/rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_P/FDR.py&quot;, line 45, in &lt;module&gt;
    ifile=open(sys.argv[1]);title=ifile.readline();
IOError: [Errno 2] No such file or directory: &#39;/tmp/tmpQ7Q7J5/JCEC_RI/rMATS_result_P-V.txt&#39;
paste: /tmp/tmpQ7Q7J5/JCEC_RI/rMATS_result_FDR.txt: No such file or directory
The statistical part is done.
</code></pre>
<p>The solution is:</p>
<p>1, install libgsl libraries</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">sudo apt-get install libgsl-dev</span><br></pre></td></tr></table></figure>

<p>2, create a link between the 16.04 library file with that expected by rMATS</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">sudo ln /usr/lib/x86_64-linux-gnu/libgsl.so.19 /usr/lib/x86_64-linux-gnu/libgsl.so.0</span><br></pre></td></tr></table></figure>

<p>3, run rMATS</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br></pre></td><td class="code"><pre><span class="line">rMATS.4.0.2/rMATS-turbo-Linux-UCS4/rMATS_C/rMATSexe</span><br><span class="line">number of thread=1; input file=input.txt; output folder=.; cutoff=0.0001;</span><br><span class="line">Total Wallclock time taken 0 seconds 0 milliseconds</span><br><span class="line">Wallclock time per thread taken 0 seconds 0 milliseconds</span><br></pre></td></tr></table></figure>
    </div>

    
    
    
      

        <div class="reward-container">
  <div>è§‰å¾—æ–‡ç« å¯¹æ‚¨æœ‰å¸®åŠ©è¯·æˆ‘å–æ¯å¥¶èŒ¶å§ğŸ˜‰</div>
  <button onclick="var qr = document.getElementById('qr'); qr.style.display = (qr.style.display === 'none') ? 'block' : 'none';">
    <i class="fa fa-qrcode fa-2x" style="line-height:35px;"></i>
  </button>
  <div id="qr" style="display: none;">
      
      <div style="display: inline-block;">
        <img src="/images/wc.png" alt="æé¹ç¨‹ å¾®ä¿¡æ”¯ä»˜">
        <p>å¾®ä¿¡æ”¯ä»˜</p>
      </div>

  </div>
</div>

        

<div>
<ul class="post-copyright">
  <li class="post-copyright-author">
    <strong>æœ¬æ–‡ä½œè€…ï¼š </strong>æé¹ç¨‹
  </li>
  <li class="post-copyright-link">
    <strong>æœ¬æ–‡é“¾æ¥ï¼š</strong>
    <a href="https://benettlee.github.io/2021/05/30/N9AxXFYknJwvXM7D/" title="RNAseqåˆ†ææµç¨‹ï¼ˆBYMEï¼‰">https://benettlee.github.io/2021/05/30/N9AxXFYknJwvXM7D/</a>
  </li>
  <li class="post-copyright-license">
    <strong>ç‰ˆæƒå£°æ˜ï¼š </strong>æœ¬åšå®¢æ‰€æœ‰æ–‡ç« é™¤ç‰¹åˆ«å£°æ˜å¤–ï¼Œå‡é‡‡ç”¨ <a href="https://creativecommons.org/licenses/by-nc-sa/4.0/" rel="noopener external nofollow noreferrer" target="_blank"><i class="fab fa-fw fa-creative-commons"></i>BY-NC-SA</a> è®¸å¯åè®®ã€‚è½¬è½½è¯·æ³¨æ˜å‡ºå¤„ï¼
  </li>
</ul>
</div>


      <footer class="post-footer">

        


        
    <div class="post-nav">
      <div class="post-nav-item">
    <a href="/2021/05/30/iyzvlY0qBaOSoVaB/" rel="prev" title="è½¬å½•ç»„æ•°æ®åˆ†ææµç¨‹">
      <i class="fa fa-chevron-left"></i> è½¬å½•ç»„æ•°æ®åˆ†ææµç¨‹
    </a></div>
      <div class="post-nav-item">
    <a href="/2021/05/30/QoG87t6VSCEerffn/" rel="next" title="KEGGè‡ªåŠ¨æ³¨é‡Šå·¥å…·KAASç®€ä»‹åŠKOæ³¨é‡Šè¡¨çš„ç®€å•å¤„ç†">
      KEGGè‡ªåŠ¨æ³¨é‡Šå·¥å…·KAASç®€ä»‹åŠKOæ³¨é‡Šè¡¨çš„ç®€å•å¤„ç† <i class="fa fa-chevron-right"></i>
    </a></div>
    </div>
      </footer>
    
  </article>
  
  
  



          </div>
          

<script>
  window.addEventListener('tabs:register', () => {
    let { activeClass } = CONFIG.comments;
    if (CONFIG.comments.storage) {
      activeClass = localStorage.getItem('comments_active') || activeClass;
    }
    if (activeClass) {
      let activeTab = document.querySelector(`a[href="#comment-${activeClass}"]`);
      if (activeTab) {
        activeTab.click();
      }
    }
  });
  if (CONFIG.comments.storage) {
    window.addEventListener('tabs:click', event => {
      if (!event.target.matches('.tabs-comment .tab-content .tab-pane')) return;
      let commentClass = event.target.classList[1];
      localStorage.setItem('comments_active', commentClass);
    });
  }
</script>

        </div>
          
  
  <div class="toggle sidebar-toggle">
    <span class="toggle-line toggle-line-first"></span>
    <span class="toggle-line toggle-line-middle"></span>
    <span class="toggle-line toggle-line-last"></span>
  </div>

  <aside class="sidebar">
    <div class="sidebar-inner">

      <ul class="sidebar-nav motion-element">
        <li class="sidebar-nav-toc">
          æ–‡ç« ç›®å½•
        </li>
        <li class="sidebar-nav-overview">
          ç«™ç‚¹æ¦‚è§ˆ
        </li>
      </ul>

      <!--noindex-->
      <div class="post-toc-wrap sidebar-panel">
          <div class="post-toc motion-element"><ol class="nav"><li class="nav-item nav-level-3"><a class="nav-link" href="#linux-%E5%B8%B8%E7%94%A8%E5%91%BD%E4%BB%A4"><span class="nav-number">1.</span> <span class="nav-text">linux å¸¸ç”¨å‘½ä»¤</span></a></li></ol></li><li class="nav-item nav-level-2"><a class="nav-link" href="#RNAseq%E7%AE%80%E5%8D%95%E5%88%86%E6%9E%90%E6%B5%81%E7%A8%8B"><span class="nav-number"></span> <span class="nav-text">RNAseqç®€å•åˆ†ææµç¨‹</span></a><ol class="nav-child"><li class="nav-item nav-level-3"><a class="nav-link" href="#install-softwares"><span class="nav-number">1.</span> <span class="nav-text">install softwares</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#raw-data%E7%9A%84%E8%BF%87%E6%BB%A4%E5%92%8C%E6%B8%85%E9%99%A4%E4%B8%8D%E5%8F%AF%E4%BF%A1%E6%95%B0%E6%8D%AE%EF%BC%88clean-reads%EF%BC%89"><span class="nav-number">2.</span> <span class="nav-text">raw dataçš„è¿‡æ»¤å’Œæ¸…é™¤ä¸å¯ä¿¡æ•°æ®ï¼ˆclean readsï¼‰</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E7%94%A8fastp%E5%AF%B9%E8%BD%AC%E5%BD%95%E7%BB%84%E6%95%B0%E6%8D%AE%E5%81%9AQC"><span class="nav-number">3.</span> <span class="nav-text">ç”¨fastpå¯¹è½¬å½•ç»„æ•°æ®åšQC</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Step1%EF%BC%9A%E5%BB%BA%E7%AB%8B%E7%B4%A2%E5%BC%95%EF%BC%8C%E6%8A%8Areads-mapping%E5%88%B0%E5%9F%BA%E5%9B%A0%E7%BB%84%E4%B8%8A"><span class="nav-number">4.</span> <span class="nav-text">Step1ï¼šå»ºç«‹ç´¢å¼•ï¼ŒæŠŠreads mappingåˆ°åŸºå› ç»„ä¸Š</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Step2%EF%BC%9A%E8%BD%AC%E5%BD%95%E6%9C%AC%E7%BB%84%E8%A3%85%E4%B8%8E%E6%8B%BC%E6%8E%A5%EF%BC%88cufflinks%E6%88%96%E8%80%85StringTie%EF%BC%89"><span class="nav-number">5.</span> <span class="nav-text">Step2ï¼šè½¬å½•æœ¬ç»„è£…ä¸æ‹¼æ¥ï¼ˆcufflinksæˆ–è€…StringTieï¼‰</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Step3%EF%BC%9A%E8%AE%A1%E7%AE%97counts%E6%95%B0"><span class="nav-number">6.</span> <span class="nav-text">Step3ï¼šè®¡ç®—countsæ•°</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#%E5%B7%AE%E5%BC%82%E8%A1%A8%E8%BE%BE%E5%88%86%E6%9E%90"><span class="nav-number">7.</span> <span class="nav-text">å·®å¼‚è¡¨è¾¾åˆ†æ</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#rMATS%E5%B7%AE%E5%BC%82%E5%8F%AF%E5%8F%98%E5%89%AA%E5%88%87%E5%88%86%E6%9E%90"><span class="nav-number">8.</span> <span class="nav-text">rMATSå·®å¼‚å¯å˜å‰ªåˆ‡åˆ†æ</span></a></li><li class="nav-item nav-level-3"><a class="nav-link" href="#Getting-error-in-running-rMATS"><span class="nav-number">9.</span> <span class="nav-text">Getting error in running rMATS</span></a></li></ol></div>
      </div>
      <!--/noindex-->

      <div class="site-overview-wrap sidebar-panel">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
    <img class="site-author-image" itemprop="image" alt="æé¹ç¨‹"
      src="/images/tx.jpg">
  <p class="site-author-name" itemprop="name">æé¹ç¨‹</p>
  <div class="site-description" itemprop="description">ä¸€å®šè¦åšæœ€é€‚åˆè‡ªå·±çš„äº‹æƒ…ï¼Œä¸è¦è¿åˆåˆ«äººçš„å£å‘³è€Œå»åšä¸€ä»¶ä¸å±äºè‡ªæˆ‘çš„â€œéš¾äº‹â€ã€‚</div>
</div>
<div class="site-state-wrap motion-element">
  <nav class="site-state">
      <div class="site-state-item site-state-posts">
          <a href="/archives/">
        
          <span class="site-state-item-count">28</span>
          <span class="site-state-item-name">æ—¥å¿—</span>
        </a>
      </div>
      <div class="site-state-item site-state-categories">
            <a href="/categories/">
          
        <span class="site-state-item-count">6</span>
        <span class="site-state-item-name">åˆ†ç±»</span></a>
      </div>
      <div class="site-state-item site-state-tags">
            <a href="/tags/">
          
        <span class="site-state-item-count">12</span>
        <span class="site-state-item-name">æ ‡ç­¾</span></a>
      </div>
  </nav>
</div>
  <div class="links-of-author motion-element">
      <span class="links-of-author-item">
        <a href="https://github.com/benettlee/benettlee.github.io" title="Github â†’ https:&#x2F;&#x2F;github.com&#x2F;benettlee&#x2F;benettlee.github.io" rel="noopener external nofollow noreferrer" target="_blank"><i class="fab fa-github fa-fw"></i>Github</a>
      </span>
      <span class="links-of-author-item">
        <a href="mailto:lipengcheng_wyyx@163.com" title="E-Mail â†’ mailto:lipengcheng_wyyx@163.com" rel="noopener external nofollow noreferrer" target="_blank"><i class="fa fa-envelope fa-fw"></i>E-Mail</a>
      </span>
      <span class="links-of-author-item">
        <a href="/images/wechat_channel.jpg" title="WeChat â†’ &#x2F;images&#x2F;wechat_channel.jpg"><i class="fab fa-weixin fa-fw"></i>WeChat</a>
      </span>
      <span class="links-of-author-item">
        <a href="/atom.xml" title="RSS â†’ &#x2F;atom.xml"><i class="fa fa-rss fa-fw"></i>RSS</a>
      </span>
  </div>



    <div class="links-of-blogroll motion-element links-of-blogroll-block">
      <div class="links-of-blogroll-title">
        <!-- modify icon to fire by szw -->
        <i class="fa fa-history fa-" aria-hidden="true"></i>
        è¿‘æœŸæ–‡ç« 
      </div>
      <ul class="links-of-blogroll-list">
        
        
          <li>
            <a href="/2021/05/30/QoG87t6VSCEerffn/" title="KEGGè‡ªåŠ¨æ³¨é‡Šå·¥å…·KAASç®€ä»‹åŠKOæ³¨é‡Šè¡¨çš„ç®€å•å¤„ç†" target="_blank">KEGGè‡ªåŠ¨æ³¨é‡Šå·¥å…·KAASç®€ä»‹åŠKOæ³¨é‡Šè¡¨çš„ç®€å•å¤„ç†</a>
          </li>
        
          <li>
            <a href="/2021/05/30/N9AxXFYknJwvXM7D/" title="RNAseqåˆ†ææµç¨‹ï¼ˆBYMEï¼‰" target="_blank">RNAseqåˆ†ææµç¨‹ï¼ˆBYMEï¼‰</a>
          </li>
        
          <li>
            <a href="/2021/05/30/iyzvlY0qBaOSoVaB/" title="è½¬å½•ç»„æ•°æ®åˆ†ææµç¨‹" target="_blank">è½¬å½•ç»„æ•°æ®åˆ†ææµç¨‹</a>
          </li>
        
          <li>
            <a href="/2021/05/30/GTrw6V1kkLax9y4L/" title="misaå’Œprimer3ç»“åˆå¿«é€Ÿè®¾è®¡SSRå¼•ç‰©" target="_blank">misaå’Œprimer3ç»“åˆå¿«é€Ÿè®¾è®¡SSRå¼•ç‰©</a>
          </li>
        
          <li>
            <a href="/2021/05/30/XuF5hPxi1nn6nSOo/" title="linuxå®‰è£…è½¯ä»¶æŠ¥é”™ä¿¡æ¯å¤„ç†æ–¹æ³•" target="_blank">linuxå®‰è£…è½¯ä»¶æŠ¥é”™ä¿¡æ¯å¤„ç†æ–¹æ³•</a>
          </li>
        
      </ul>
    </div>



<div style="">
  <canvas id="canvas" style="width:60%;">å½“å‰æµè§ˆå™¨ä¸æ”¯æŒcanvasï¼Œè¯·æ›´æ¢æµè§ˆå™¨åå†è¯•</canvas>
</div>
<script>
(function(){

   var digit=
    [
        [
            [0,0,1,1,1,0,0],
            [0,1,1,0,1,1,0],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [0,1,1,0,1,1,0],
            [0,0,1,1,1,0,0]
        ],//0
        [
            [0,0,0,1,1,0,0],
            [0,1,1,1,1,0,0],
            [0,0,0,1,1,0,0],
            [0,0,0,1,1,0,0],
            [0,0,0,1,1,0,0],
            [0,0,0,1,1,0,0],
            [0,0,0,1,1,0,0],
            [0,0,0,1,1,0,0],
            [0,0,0,1,1,0,0],
            [1,1,1,1,1,1,1]
        ],//1
        [
            [0,1,1,1,1,1,0],
            [1,1,0,0,0,1,1],
            [0,0,0,0,0,1,1],
            [0,0,0,0,1,1,0],
            [0,0,0,1,1,0,0],
            [0,0,1,1,0,0,0],
            [0,1,1,0,0,0,0],
            [1,1,0,0,0,0,0],
            [1,1,0,0,0,1,1],
            [1,1,1,1,1,1,1]
        ],//2
        [
            [1,1,1,1,1,1,1],
            [0,0,0,0,0,1,1],
            [0,0,0,0,1,1,0],
            [0,0,0,1,1,0,0],
            [0,0,1,1,1,0,0],
            [0,0,0,0,1,1,0],
            [0,0,0,0,0,1,1],
            [0,0,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [0,1,1,1,1,1,0]
        ],//3
        [
            [0,0,0,0,1,1,0],
            [0,0,0,1,1,1,0],
            [0,0,1,1,1,1,0],
            [0,1,1,0,1,1,0],
            [1,1,0,0,1,1,0],
            [1,1,1,1,1,1,1],
            [0,0,0,0,1,1,0],
            [0,0,0,0,1,1,0],
            [0,0,0,0,1,1,0],
            [0,0,0,1,1,1,1]
        ],//4
        [
            [1,1,1,1,1,1,1],
            [1,1,0,0,0,0,0],
            [1,1,0,0,0,0,0],
            [1,1,1,1,1,1,0],
            [0,0,0,0,0,1,1],
            [0,0,0,0,0,1,1],
            [0,0,0,0,0,1,1],
            [0,0,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [0,1,1,1,1,1,0]
        ],//5
        [
            [0,0,0,0,1,1,0],
            [0,0,1,1,0,0,0],
            [0,1,1,0,0,0,0],
            [1,1,0,0,0,0,0],
            [1,1,0,1,1,1,0],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [0,1,1,1,1,1,0]
        ],//6
        [
            [1,1,1,1,1,1,1],
            [1,1,0,0,0,1,1],
            [0,0,0,0,1,1,0],
            [0,0,0,0,1,1,0],
            [0,0,0,1,1,0,0],
            [0,0,0,1,1,0,0],
            [0,0,1,1,0,0,0],
            [0,0,1,1,0,0,0],
            [0,0,1,1,0,0,0],
            [0,0,1,1,0,0,0]
        ],//7
        [
            [0,1,1,1,1,1,0],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [0,1,1,1,1,1,0],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [0,1,1,1,1,1,0]
        ],//8
        [
            [0,1,1,1,1,1,0],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [1,1,0,0,0,1,1],
            [0,1,1,1,0,1,1],
            [0,0,0,0,0,1,1],
            [0,0,0,0,0,1,1],
            [0,0,0,0,1,1,0],
            [0,0,0,1,1,0,0],
            [0,1,1,0,0,0,0]
        ],//9
        [
            [0,0,0,0,0,0,0],
            [0,0,1,1,1,0,0],
            [0,0,1,1,1,0,0],
            [0,0,1,1,1,0,0],
            [0,0,0,0,0,0,0],
            [0,0,0,0,0,0,0],
            [0,0,1,1,1,0,0],
            [0,0,1,1,1,0,0],
            [0,0,1,1,1,0,0],
            [0,0,0,0,0,0,0]
        ]//:
    ];

var canvas = document.getElementById('canvas');

if(canvas.getContext){
    var cxt = canvas.getContext('2d');
    //å£°æ˜canvasçš„å®½é«˜
    var H = 100,W = 700;
    canvas.height = H;
    canvas.width = W;
    cxt.fillStyle = '#f00';
    cxt.fillRect(10,10,50,50);

    //å­˜å‚¨æ—¶é—´æ•°æ®
    var data = [];
    //å­˜å‚¨è¿åŠ¨çš„å°çƒ
    var balls = [];
    //è®¾ç½®ç²’å­åŠå¾„
    var R = canvas.height/20-1;
    (function(){
        var temp = /(\d)(\d):(\d)(\d):(\d)(\d)/.exec(new Date());
        //å­˜å‚¨æ—¶é—´æ•°å­—ï¼Œç”±åä½å°æ—¶ã€ä¸ªä½å°æ—¶ã€å†’å·ã€åä½åˆ†é’Ÿã€ä¸ªä½åˆ†é’Ÿã€å†’å·ã€åä½ç§’é’Ÿã€ä¸ªä½ç§’é’Ÿè¿™7ä¸ªæ•°å­—ç»„æˆ
        data.push(temp[1],temp[2],10,temp[3],temp[4],10,temp[5],temp[6]);
    })();

    /*ç”Ÿæˆç‚¹é˜µæ•°å­—*/
    function renderDigit(index,num){
        for(var i = 0; i < digit[num].length; i++){
            for(var j = 0; j < digit[num][i].length; j++){
                if(digit[num][i][j] == 1){
                    cxt.beginPath();
                    cxt.arc(14*(R+2)*index + j*2*(R+1)+(R+1),i*2*(R+1)+(R+1),R,0,2*Math.PI);
                    cxt.closePath();
                    cxt.fill();
                }
            }
        }
    }

    /*æ›´æ–°æ—¶é’Ÿ*/
    function updateDigitTime(){
        var changeNumArray = [];
        var temp = /(\d)(\d):(\d)(\d):(\d)(\d)/.exec(new Date());
        var NewData = [];
        NewData.push(temp[1],temp[2],10,temp[3],temp[4],10,temp[5],temp[6]);
        for(var i = data.length-1; i >=0 ; i--){
            //æ—¶é—´å‘ç”Ÿå˜åŒ–
            if(NewData[i] !== data[i]){
                //å°†å˜åŒ–çš„æ•°å­—å€¼å’Œåœ¨dataæ•°ç»„ä¸­çš„ç´¢å¼•å­˜å‚¨åœ¨changeNumArrayæ•°ç»„ä¸­
                changeNumArray.push(i+'_'+(Number(data[i])+1)%10);
            }
        }
        //å¢åŠ å°çƒ
        for(var i = 0; i< changeNumArray.length; i++){
            addBalls.apply(this,changeNumArray[i].split('_'));
        }
        data = NewData.concat();
    }

    /*æ›´æ–°å°çƒçŠ¶æ€*/
    function updateBalls(){
        for(var i = 0; i < balls.length; i++){
            balls[i].stepY += balls[i].disY;
            balls[i].x += balls[i].stepX;
            balls[i].y += balls[i].stepY;
            if(balls[i].x > W + R || balls[i].y > H + R){
                balls.splice(i,1);
                i--;
            }
        }
    }

    /*å¢åŠ è¦è¿åŠ¨çš„å°çƒ*/
    function addBalls(index,num){
        var numArray = [1,2,3];
        var colorArray =  ["#3BE","#09C","#A6C","#93C","#9C0","#690","#FB3","#F80","#F44","#C00"];
        for(var i = 0; i < digit[num].length; i++){
            for(var j = 0; j < digit[num][i].length; j++){
                if(digit[num][i][j] == 1){
                    var ball = {
                        x:14*(R+2)*index + j*2*(R+1)+(R+1),
                        y:i*2*(R+1)+(R+1),
                        stepX:Math.floor(Math.random() * 4 -2),
                        stepY:-2*numArray[Math.floor(Math.random()*numArray.length)],
                        color:colorArray[Math.floor(Math.random()*colorArray.length)],
                        disY:1
                    };
                    balls.push(ball);
                }
            }
        }
    }

    /*æ¸²æŸ“*/
    function render(){
        //é‡ç½®ç”»å¸ƒå®½åº¦ï¼Œè¾¾åˆ°æ¸…ç©ºç”»å¸ƒçš„æ•ˆæœ
        canvas.height = 100;
        //æ¸²æŸ“æ—¶é’Ÿ
        for(var i = 0; i < data.length; i++){
            renderDigit(i,data[i]);
        }
        //æ¸²æŸ“å°çƒ
        for(var i = 0; i < balls.length; i++){
            cxt.beginPath();
            cxt.arc(balls[i].x,balls[i].y,R,0,2*Math.PI);
            cxt.fillStyle = balls[i].color;
            cxt.closePath();
            cxt.fill();
        }
    }

    clearInterval(oTimer);
    var oTimer = setInterval(function(){
        //æ›´æ–°æ—¶é’Ÿ
        updateDigitTime();
        //æ›´æ–°å°çƒçŠ¶æ€
        updateBalls();
        //æ¸²æŸ“
        render();
    },50);
}

})();
</script>

      </div>

    </div>
  </aside>
  <div id="sidebar-dimmer"></div>


      </div>
    </main>

    <footer class="footer">
      <div class="footer-inner">
        

        

<div class="copyright">
  
  &copy; 2019 â€“ 
  <span itemprop="copyrightYear">2021</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">æé¹ç¨‹</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-chart-area"></i>
    </span>
    <span title="ç«™ç‚¹æ€»å­—æ•°">614k</span>
    <span class="post-meta-divider">|</span>
    <span class="post-meta-item-icon">
      <i class="fa fa-coffee"></i>
    </span>
    <span title="ç«™ç‚¹é˜…è¯»æ—¶é•¿">9:18</span>
</div><div class="translate-style">
ç¹/ç®€ï¼š<a id="translateLink" href="javascript:translatePage();" rel="external nofollow noreferrer">ç¹ä½“
</a>
</div>
<script type="text/javascript" src="/js/tw_cn.js"></script>
<script type="text/javascript">
var defaultEncoding = 2; //ç½‘ç«™ç¼–å†™å­—ä½“æ˜¯å¦ç¹ä½“ï¼Œ1-ç¹ä½“ï¼Œ2-ç®€ä½“
var translateDelay = 0; //å»¶è¿Ÿæ—¶é—´,è‹¥ä¸åœ¨å‰, è¦è®¾å®šå»¶è¿Ÿç¿»è¯‘æ—¶é—´, å¦‚100è¡¨ç¤º100ms,é»˜è®¤ä¸º0
var cookieDomain = "https://tding.top/"; //Cookieåœ°å€, ä¸€å®šè¦è®¾å®š, é€šå¸¸ä¸ºä½ çš„ç½‘å€
var msgToTraditionalChinese = "ç¹ä½“"; //æ­¤å¤„å¯ä»¥æ›´æ”¹ä¸ºä½ æƒ³è¦æ˜¾ç¤ºçš„æ–‡å­—
var msgToSimplifiedChinese = "ç®€ä½“"; //åŒä¸Šï¼Œä½†ä¸¤å¤„å‡ä¸å»ºè®®æ›´æ”¹
var translateButtonId = "translateLink"; //é»˜è®¤äº’æ¢id
translateInitilization();
</script>


        








      </div>
    </footer>
  </div>

  
  <script src="/lib/anime.min.js"></script>
  <script src="/lib/velocity/velocity.min.js"></script>
  <script src="/lib/velocity/velocity.ui.min.js"></script>

<script src="/js/utils.js"></script>

<script src="/js/motion.js"></script>


<script src="/js/schemes/pisces.js"></script>


<script src="/js/next-boot.js"></script>




  




  
<script src="/js/local-search.js"></script>













  

  
  <script src="//cdn.jsdelivr.net/npm/quicklink@1/dist/quicklink.umd.js"></script>
  <script>
      window.addEventListener('load', () => {
      quicklink({
        timeout : 3000,
        priority: true,
        ignores : [uri => uri.includes('#'),uri => uri === 'https://benettlee.github.io/2021/05/30/N9AxXFYknJwvXM7D/',]
      });
      });
  </script>

</body>
</html>
